### PR TITLE
prevent dolt filter branch when it would overwrite unchecked branch's working set

### PR DESCRIPTION
**Pull request by jycor**
Turns out other branches can have working sets, and dolt-filter branch would drop those. PR prevents that from happening.

adding tests to comments I missed here:
https://github.com/dolthub/dolt/pull/7895

### CODE REVIEW

**Comment by coffeegoddd** [comment]
@coffeegoddd __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>eafc6df</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>eafc6df</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

File: integration-tests/bats/filter-branch.bats [hunk 1]
```diff
@@ -427,3 +427,88 @@ SQL
     [[ "$output" =~ "| 3  | 3  | NULL   |" ]] || false
     [[ "$output" =~ "+----+----+--------+" ]] || false
 }
+
+@test "filter-branch: --all fails with working and staged changes" {
```
Comment (timsehn) [inline integration-tests/bats/filter-branch.bats:431]: I feel like you actually have to make a branch and have the working set on the other branch to test `--branches` and `--all`

**Comment by coffeegoddd** [comment]
@jycor __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>f60394f</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>f60394f</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@coffeegoddd __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>f8a4f3a</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>f8a4f3a</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Review by timsehn — APPROVED** [review]
Tests look good now :-)

File: integration-tests/bats/filter-branch.bats [hunk 2]
```diff
@@ -427,3 +427,122 @@ SQL
...
+    dolt sql << SQL
+call dolt_checkout('-b', 'other');
+insert into test values (3, 3);
+SQL
+    run dolt filter-branch -q "alter table test add column filter int"
+    [ "$status" -eq 0 ]
+
+    run dolt sql -q "select * from test as of 'HEAD'"
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "+----+----+--------+" ]] || false
```
Comment (zachmu) [inline integration-tests/bats/filter-branch.bats:441]: This format is pretty fragile, you should use the `-r csv` option to check row output from `dolt sql`

Comment (zachmu) [reply integration-tests/bats/filter-branch.bats:441]: Same comment in other places

**Review by zachmu — APPROVED** [review]
LGTM

**Comment by coffeegoddd** [comment]
@jycor __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>0f54223</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>0f54223</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Pull request merged**